circuits/scripts/withdraw.js

```diff
@@ -0,0 +1,130 @@
+import { expect, assert } from 'chai';
+import { ProjectivePoint, signAsync } from '@noble/secp256k1';
+import { default as vmtree } from 'vmtree-sdk';
+
+import { MerkleTree as FixedMerkleTree } from "fixed-merkle-tree";
+
+class MerkleTree extends FixedMerkleTree {
+    constructor({ hasher, levels = 20, leaves = [], zero = 0 }) {
+        super(levels, leaves, {
+            hashFunction: (left, right) => hasher([left, right]),
+            zeroElement: zero,
+        });
+    };
+};
+
+const stateTree = new MerkleTree({ hasher: vmtree.poseidon, levels: 3, zero: 0 });
+
+function bigint_to_tuple(x/*: bigint*/) {
+    let mod/*: bigint*/ = 2n ** 64n;
+    let ret = [0n, 0n, 0n, 0n];
+
+    var x_temp = x;
+    for (var idx = 0; idx < ret.length; idx++) {
+        ret[idx] = x_temp % mod;
+        x_temp = x_temp / mod;
+    }
+    return ret;
+}
+
+function bigint_to_array(n/*: number*/, k/*: number*/, x/*: bigint*/) {
+    let mod = 1n;
+    for (var idx = 0; idx < n; idx++) {
+        mod = mod * 2n;
+    }
+
+    let ret = [];
+    var x_temp = x;
+    for (var idx = 0; idx < k; idx++) {
+        ret.push(x_temp % mod);
+        x_temp = x_temp / mod;
+    }
+    return ret;
+}
```

Comment:
Let's import from dependencies rather than copying the code (copied from [here](https://github.com/0xPARC/circom-ecdsa/blob/d87eb7068cb35c951187093abe966275c1839ead/test/ecdsa.test.ts#L13-L38) right?). It just keeps the repo cleaner and easier to fully understand.

Reply:
there is some complexity here. importing ts function in a js file. we can later convert these js files into ts, but in general i support your comment.

Reply:
Good point. No rush to tidy this up.